sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -241,6 +241,7 @@ trait RuntimeReplaceable extends UnaryExpression with Unevaluable {
   override def nullable: Boolean = child.nullable
   override def foldable: Boolean = child.foldable
   override def dataType: DataType = child.dataType
+  override lazy val canonicalized: Expression = child.canonicalized
```

Comment:
First, my only concern is `CleanupAliases` might be called in the other Analyzer rules. These unneeded aliases could break the assumptions made by the caller of `CleanupAliases`.

Second, you should post what I said above in the PR description. So far, the description is not clear to the reviewers.  

Third, I am ok about the fix, but we should write the comments to explain why we did it here. We need to add two comments. One is in `CleanupAliases `; another is above this line to explain why we add `canonicalized`.